### PR TITLE
[Model] Add Pacific-Prime / Complexity model support

### PR DESCRIPTION
**Pull request by Complexity-ML**
## Purpose

Add support for the **Pacific-Prime** (Complexity) model family, a 1.5B parameter language model with novel architectural innovations.

Closes #32595

**Model**: https://huggingface.co/Pacific-Prime/pacific-prime
**Paper**: https://zenodo.org/records/18293026

### Key Innovations
- **INL Dynamics**: PID-like control with velocity tracking for numerical stability
- **Token-Routed MLP**: Deterministic 4-expert MoE routing based on token ID
- **Mu-Guided Attention**: Top-down influence from equilibrium state on Q/K/V
- **GQA**: 16 heads / 8 KV heads

## Test Plan

```bash
# Start server
vllm serve Pacific-Prime/pacific-prime --dtype float16 --max-model-len 2048

# Test inference
curl http://localhost:8080/v1/completions \
  -H "Content-Type: application/json" \
  -d '{"model": "Pacific-Prime/pacific-prime", "prompt": "Hello", "max_tokens": 50}'
  
  
  Test Result
Tested successfully on NVIDIA RTX 5090 32GB, CUDA 12.9, vLLM 0.14.0:

Model loads correctly ✓
Inference produces expected outputs ✓
Memory-efficient MoE implementation ✓
OpenAI-compatible API works ✓

### CODE REVIEW

File: vllm/model_executor/models/pacific_prime.py [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+        input_ids: Optional[torch.Tensor],
+        positions: torch.Tensor,
+        inputs_embeds: Optional[torch.Tensor] = None,
+        intermediate_tensors: Optional[IntermediateTensors] = None,
+    ) -> torch.Tensor:
+        if inputs_embeds is not None:
+            hidden_states = inputs_embeds
+        else:
+            hidden_states = self.embed_tokens(input_ids)
+        velocity_states = torch.zeros_like(hidden_states)
```
[the remark this answers is not in the document]
Comment (Complexity-ML) [reply vllm/model_executor/models/pacific_prime.py:548]: This is intentional. velocity_states track dynamics within a single forward pass (across layers), not across different inference calls. Each request starts fresh, similar to hidden_states. The INL dynamics stabilize the signal propagation through the 24 layers during a single generation step.

File: vllm/model_executor/models/pacific_prime.py [hunk 2]
```diff
@@ -0,0 +1,715 @@
...
+        num_experts: int,
+        vocab_size: int,
+        quant_config: Optional[QuantizationConfig] = None,
+        prefix: str = "",
+    ):
+        super().__init__()
+        self.hidden_size = hidden_size
+        self.num_experts = num_experts
+        self.vocab_size = vocab_size
+        self.expert_intermediate_size = intermediate_size // num_experts
```
[the remark this answers is not in the document]
Comment (Complexity-ML) [reply vllm/model_executor/models/pacific_prime.py:139]: Good point. The model config guarantees divisibility (5632 / 4 = 1408). I can add an assert for safety:
assert intermediate_size % num_experts == 0, "intermediate_size must be divisible by num_experts"

File: vllm/model_executor/models/pacific_prime.py [hunk 3]
```diff
@@ -0,0 +1,715 @@
...
+            gate_up_out = x_expert @ gate_up_w  # [n, 2*intermediate]
+            gate_out = gate_up_out[..., :self.expert_intermediate_size]
+            up_out = gate_up_out[..., self.expert_intermediate_size:]
+            intermediate = F.silu(gate_out) * up_out  # [n, intermediate]
+            expert_output = intermediate @ down_w  # [n, hidden_size]
+
+            # Scatter back to output
+            output[mask] = expert_output
+
+        return output
```
[the remark this answers is not in the document]
Comment (Complexity-ML) [reply vllm/model_executor/models/pacific_prime.py]: This is a deliberate memory-vs-speed tradeoff. The previous per-token bmm implementation caused OOM on 32GB GPUs. With only 4 experts, the loop overhead is minimal. For production use with more experts, a fused kernel (like Triton) would be better, but this keeps the PR simple and functional. Happy to optimize in a follow-up PR if there's interest.
